libcalico-go/lib/ipam/ipam_block_reader_writer.go

```diff
@@ -153,7 +153,7 @@ func (rw blockReaderWriter) findUsableBlock(ctx context.Context, host string, ve
 			if info, ok := exists[subnet.String()]; !ok {
 				log.Infof("Found free block: %+v", *subnet)
 				return subnet, nil
-			} else if info.affinity == host && info.numFree != 0 {
+			} else if info.affinity == affinityCfg.Host && info.numFree != 0 {
```

Comment:
I think this check needs to also check that the affinity type is the same, otherwise we encounter the same kind of bugs where a host with name `virtual-loadbalancer` will confuse our code.

The purpose of passing an `AffinityConfig` down through these functions is so that they can be aware of the `Type` field.